aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -919,6 +919,41 @@ Tensor _nested_view_from_buffer(
     std::vector<int64_t>(offsets.begin(), offsets.end()));
 }
 
+/**
+ * Create a [B, *, D] nested tensor that is a view of a 2D values tensor
+ * where offsets are the cumsum of the size along the ragged dimension
+ *
+ * @return Returns a nested tensor that
+ * aliases the same storage as values
+ */
+Tensor _nested_from_values_and_offsets(
+    const Tensor& values,
+    const Tensor& offsets) {
+  TORCH_CHECK(values.dim() == 2, "values must be a 2D tensor, got values.dim() ", values.dim(), ".");
+  TORCH_CHECK(values.is_contiguous(), "values must be contiguous.");
+  TORCH_CHECK(offsets.dim() == 1, "offsets must be a 1D tensor, got offsets.dim() ", offsets.dim(), ".");
+
+  TORCH_CHECK(offsets.select(0, 0).item<int64_t>() == 0,
```

Comment:
Could be wrong and we may want to measure/ may not be worth. I have always found that if you are working with a cpu tensor and are going to access its elements it is always substantially faster going to the data pointer first than indexing. It maybe less safe however but might be worth it for speed